File: scala/spark-utils/src/main/scala/hash32.scala
package org.dsv.spark.utils
import org.apache.spark.sql.api.java.UDF1
import org.apache.commons.codec.digest.MurmurHash3

class hash32 extends UDF1[String, Int] {

  override def call(Str: String): Int = {
    MurmurHash3.hash32x86(Str.getBytes())
  }
}

File: scala/spark-utils/src/main/scala/hash32.java
package org.dsv.spark.utils;
import org.apache.commons.codec.digest.MurmurHash3;
import java.nio.charset.StandardCharsets;
import org.apache.spark.sql.api.java.UDF1;


public class hash32 implements UDF1<String, Integer>{
    private static final long serialVersionUID = 1L;
    @Override
    public Integer call(String str) throws Exception {
        if (str == null) return (null);
        byte[] data = str.getBytes(StandardCharsets.UTF_8);
        return (MurmurHash3.hash32x86(data));
    }
}
